langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```diff
@@ -0,0 +1,150 @@
+package dev.langchain4j.store.embedding.alloydb;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Logger;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+import dev.langchain4j.store.embedding.EmbeddingSearchRequest;
+import dev.langchain4j.store.embedding.EmbeddingSearchResult;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+
+public class AlloyDBEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+    private static final Logger log = Logger.getLogger(AlloyDBEmbeddingStore.class.getName());
+    private final AlloyDBEngine engine;
+    private final String tableName;
+    private String contentColumn;
+    private String embeddingColumn;
```

Comment:
We should also have an idColumn (String), metadata_json_column (String), and schema_name (string, defaults to public)